clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -355,7 +362,70 @@ func (pc *Controller) applyManifests(path, manifestGlob string) error {
 		os.DirFS(path), manifestGlob, pc.templateMapping, client.Retry(apierrs.IsNotFound))
 }
 
-// exposeAPIServerMetrics configures anonymous access to the apiserver metrics.
+// create ServiceAccount and token used by Prometheus to fetch data from testing cluster
+func (pc *Controller) createServiceAccountInTestingCluster(k8sClient kubernetes.Interface) error {
+	klog.V(2).Info("Creating ServiceAccount in testing cluster")
+	// We need to get a client to the cluster where the test is being executed on,
+	// not the cluster that the prometheus is running in. Usually, there is only
+	// once cluster, but in case of kubemark we have two and thus we need to
+	// create a new client here.
+	clientSet, err := framework.NewMultiClientSet(
```

Comment:
We already create the test cluster's clientset in the `exposeAPIServerMetrics` method, so we're duplicating this logic here.

I propose creating that clientset at the beginning of the `SetUpPrometheusStack` method and reusing it in other places in this file, including this one.